chat-plugins/blackjack.js

```diff
@@ -0,0 +1,501 @@
+			let user = Users(this.spectators[i]);
+			if (user) user.sendTo(this.id, message);
+		}
+	}
+	clear() {
+		const player = this.players[this.curUser];
+		if (!player) return; // this should never happen
+		player.sendRoom(`|uhtmlchange|user-blackjack-${this.blackjackNumber}|`);
+	}
+
+	/**
+	 * Game State Changes
+	 * start - starts the game
+	 * end - ends the game
+	 * destroy - destroys the game
+	 */
+	start(user) {
+		if (Object.keys(this.players).length < this.minimumPlayers) {
+			if (this.autostart) {
+				clearTimeout(this.autostart);
+				this.autostart = null;
+			}
+			this.add("<br />Not enough players to start; game canceled.");
+			this.destroy();
+			return;
+		}
+		if (user) this.startedBy = user.name;
+		this.add(`[Blackjack has started. ${this.spectateButton}]`, true, true);
+
+		this.curUser = Object.keys(this.players)[0];
+
+		let output = `The game of blackjack has started${(this.startedBy !== '' ? ` (started by ${this.startedBy})` : ``)}.<br />`;
+		this.started = true;
+		this.state = 'started';
+		for (let player in this.players) {
+			this.giveCard(this.players[player]);
+			this.giveCard(this.players[player]);
+			output += Chat.html`<strong>${this.players[player].name}</strong>: [${this.players[player].cards[0]}] [${this.players[player].cards[1]}] (${this.players[player].points})<br />`;
+		}
+
+		this.giveCard('dealer');
+		this.giveCard('dealer');
+		output += `<strong>${this.dealer.name}</strong>: [${this.dealer.cards[0]}]`;
+
+		this.display(output, true);
+		this.next();
+	}
+	end(user, cmd) {
+		const force = (cmd && cmd === 'forceend');
+		if (cmd && !force) return this.errorMessage(user, `You can only end this game by using /blackjack forceend.`);
+		if (force) {
+			this.add(Chat.html`(Blackjack was forcibly ended by ${user.name}.)`, true);
+			if (this.curUser !== "") this.players[this.curUser].sendRoom(`|uhtmlchange|user-blackjack-${this.blackjackNumber}|`);
+		}
+		let winners = [];
+
+		if (!force) {
+			if (this.dealer.points > 21) {
+				for (let u in this.players) {
+					if (this.players[u].status === 'bust') continue;
+					winners.push(this.players[u].name);
+				}
+			} else if (this.dealer.points !== 21) {
+				for (let u in this.players) {
+					if (this.players[u].status === 'bust' || this.players[u].points <= this.dealer.points) continue;
+					winners.push(this.players[u].name);
+				}
+			} else if (this.dealer.points === 21) {
+				winners.push(this.dealer.name);
+			}
+			this.add(`[Blackjack has ended.]`, true, true, true);
+			if (winners.length < 1) {
+				this.display(`<br />There are no winners this time.`);
+			} else {
+				this.display(Chat.html`<br />Winner${Chat.plural(winners.length)}: ${winners.join(', ')}`);
+			}
+		}
+
+		if (this.autostart) {
+			clearTimeout(this.autostart);
+			this.autostart = null;
+		}
+		this.state = 'ended';
+
+		this.destroy();
+	}
+	destroy() {
+		for (let i in this.players) {
+			this.players[i].destroy();
+		}
+		this.room.blackjack++;
```

Comment:
They do, but `.poll` should probably be defined in `ChatRoom`'s constructor since some of its methods use its value. None use `.blackjack`'s value